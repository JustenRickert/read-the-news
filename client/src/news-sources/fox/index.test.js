const puppeteer = require('puppeteer')

const { collectArticle } = require('../')

describe('fox snapshots', () => {
  jest.setTimeout(30e3)

  let browser
  let page

  beforeAll(async () => {
    browser = await puppeteer.launch()
  })

  beforeEach(async () => {
    page = await browser.newPage()
  })

  afterEach(async () => {
    await page.close()
  })

  afterAll(async () => {
    await browser.close()
  })

  it('gets it good on a first pager', async () => {
    const result = await collectArticle(
      page,
      'https://www.foxnews.com/politics/impeachment-articles-senate-trial-republicans'
    )
    expect(result).toMatchSnapshot()
  })

  it('does gutfield fuuuu', async () => {
    const result = await collectArticle(
      page,
      'https://www.foxnews.com/opinion/gutfeld-on-last-nights-debate'
    )
    expect(result).toMatchSnapshot()
  })

  it('does dates statically when running in test', async () => {
    const result = await collectArticle(
      page,
      'https://www.foxnews.com/politics/trump-administration-begins-sending-migrants-to-guatemala-as-part-of-safe-third-country-agreement'
    )
    expect(result).toMatchSnapshot()
  })

  it('gets rid of ^WATCH: type advertisements', async () => {
    const result = await collectArticle(
      page,
      'https://www.foxbusiness.com/markets/charles-schwab-buying-td-ameritrade-for-26b'
    )
    expect(result).toMatchSnapshot()
  })
})
