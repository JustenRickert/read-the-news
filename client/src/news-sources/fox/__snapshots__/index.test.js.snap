// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fox snapshots does dates statically when running in test 1`] = `
Object {
  "authors": Array [
    Object {
      "href": "https://www.foxnews.com/person/s/adam-shaw",
      "name": "Adam Shaw",
    },
  ],
  "content": "The Trump administration on Thursday announced that it has started to send migrants to Guatemala as part of its “safe third country” agreement with the Central American nation — a boost for the administration’s effort to enlist other countries' help dealing with the migrant crisis.
An initial flight landed in Guatemala Thursday morning. The move is part of a “phased-in” approach to sending migrants mainly from Honduras and El Salvador to the Central American country, having established that its asylum policies meet the criteria of having a “full and fair” asylum system.
The agreement, negotiated earlier this year, is part of a series of regional agreements with countries including Mexico and El Salvador.
“We continue to work on asylum agreement implementation plans with all three Central American countries,” acting DHS Secretary Chad Wolf said in a statement to Fox News.
“For Guatemala specifically, we have begun a phased-in implementation approach, which includes sending an initial flight,” he said. “[The Department of Homeland Security], alongside the State Department, will continue to work with the Government of Guatemala and international organizations as we expand the program in the weeks and months ahead.”
DHS would not confirm operational details on how many migrants were on the plane, but a DHS official told Fox News that being moved to Guatemala was an option for migrants and that no one would be sent to the country against their will.
Migrants who are seeking asylum at the southern border will be given the option to be sent to Guatemala, sent back to their home country, or to enter the Migrant Protection Protocols (MPP) and wait in Mexico.
Migrant rights and humanitarian groups reacted with horror to the move, saying that it will potentially send migrants into a country marred by violence. The U.N. refugee agency said in a statement this week that it has “serious concerns” about the policy.
“It is an approach at variance with international law that could result in the transfer of highly vulnerable individuals to countries where they may face life-threatening dangers,” the statement said.
“UNHCR is not party to any of the bilateral Asylum Cooperative Agreements (ACA) concluded in recent months between the United States and the governments of Guatemala, Honduras and El Salvador. We are in dialogue with all of the governments concerned to enumerate our concerns as we also seek to learn more about their plans for implementation.”
But for the administration, which has prioritized securing America’s border and specifically ending the pull factors that have attracted more than a million migrants to make the journey north and attempt to enter the U.S., it’s a necessary move. The U.S. is also underwriting Guatemala’s asylum system, both through funding directly to the country and also via funding to the U.N. High Commissioner for Refugees.
The Guatemala agreement will work in conjunction with MPP, known colloquially as the “Remain-in-Mexico” policy, by which migrants seeking asylum are sent back to Mexico to await their immigration proceedings. That policy aims in part to speed up claims and also end the “catch-and-release” practice whereby migrants were released into the interior.
The administration’s approach, which includes building a wall at the southern border, has coincided with apprehensions at the border dropping by approximately 70 percent since May.",
  "href": "https://www.foxnews.com/politics/trump-administration-begins-sending-migrants-to-guatemala-as-part-of-safe-third-country-agreement",
  "publicationDate": "Mon Nov 18 2019 20:32:01 GMT-0600 (Central Standard Time)",
  "title": "Trump administration begins sending migrants to Guatemala as part of ‘safe third country’ agreement",
}
`;

exports[`fox snapshots does gutfield fuuuu 1`] = `
Object {
  "authors": Array [
    Object {
      "href": "https://www.foxnews.com/person/g/greg-gutfeld",
      "name": "Greg Gutfeld",
    },
  ],
  "content": "The clown car rolled into Atlanta to drop off a pile of broken toys, minus the batteries Wednesday night for the debate of Democratic presidential candidates.
There was \\"gramps,\\" otherwise known as former Vice President Joe Biden, who sounded like a 10-year-old coming down from laughing gas after getting a tooth pulled.
Maybe it's not his fault. He did just find out he was a grandfather!
Then there was the cop, Sen. Kamala Harris, D-Calif., pretending to be something she's not.
\\"I think that it's unfortunate that we have someone on this stage who is attempting to be the Democratic nominee for president of the United States who, during the Obama administration, spent four years full-time on Fox News criticizing President Obama,\\" Harris said.
Yeah, it's unfortunate Rep. Tulsi Gabbard, D-Hawaii, had the guts to go on TV and face tough questions. Maybe you should throw her in jail.
And who's Tom Steyer again? Did someone leave John Kerry out in the sun?
I miss Marianne Williamson and Rep. Eric Swalwell, D-Calif. He would've been a gas.
Thank goodness the candidates who did show up focused on child welfare! (In case you forgot, for Democrats, abortion is the ultimate childcare. There's no need for a babysitter when there's no baby to sit.)
The bottom line? You could have found more fire in a bag of prunes.
Impeachment theater drained them of their energy and intelligence, leaving them all searching for something to say to each other, other than \\"orange man bad.\\" None of them laid a glove on the others, which is good hygiene.
But you know what the debate stage really was?
It was that cabin at Camp Crystal Lake from \\"Friday the 13th.\\" All those candidates are the stupid teenagers making idiotic decisions. Let's check out the basement! Let's go skinny-dipping alone! Above all, don’t hop in that car and escape.
And President Trump is the ever-present Jason, waiting to pick each and every one of them off. He’s there, on that stage, even when he isn’t.
It's sad to think that someday Trump will not be president, and the whole world will look like a Democratic debate. Like Jason, maybe we’ll just keep bringing him back.",
  "href": "https://www.foxnews.com/opinion/gutfeld-on-last-nights-debate",
  "publicationDate": "Tue Nov 19 2019 20:32:01 GMT-0600 (Central Standard Time)",
  "title": "Gutfeld on Wednesday night’s debate",
}
`;

exports[`fox snapshots gets it good on a first pager 1`] = `
Object {
  "authors": Array [
    Object {
      "href": "https://www.foxnews.com/person/r/gregg-re",
      "name": "Gregg Re",
    },
    Object {
      "href": "https://www.foxnews.com/person/p/chad-pergram",
      "name": "Chad Pergram",
    },
  ],
  "content": "Abuse of power. Bribery. Contempt of Congress. Obstruction of justice.
Those are the four potential articles of impeachment that House Judiciary Committee Democrats could draw up against President Trump as soon as next month, Fox News is told, after all scheduled public hearings before the House Intelligence Committee wrapped up on a testy note Thursday.
At a meeting with top GOP senators and Trump administration officials at the White House on Thursday afternoon, Fox News is told there was a consensus that should Trump be impeached by the House, the GOP-controlled Senate should hold a full trial, rather than ignore the issue.
Reports have surfaced that Republicans were considering even holding a long trial to disrupt the 2020 presidential primaries. Several Democrats seeking to unseat Trump -- including Kamala Harris, Amy Klobuchar, Elizabeth Warren and Bernie Sanders -- are senators who would need to divert at least some of their campaigning time toward a potential trial.
\\"I think most everybody agreed there's not 51 votes to dismiss it before the managers get to call their case,\\" Senate Judiciary Committee Chairman Lindsey Graham, R-S.C., told Fox News after huddling with other top Republican senators and White House officials. \\"The idea you would dismiss the trial before they presented the cases is a non-starter. You're not going to get a motion to dismiss.\\"
It remained possible the House Intelligence Committee could schedule more hearings, although no additional hearings are expected during Thanksgiving week. Or, the committee could prepare a report on its findings for the House Judiciary Committee -- which would have the option of holding its own hearings or simply drafting articles of impeachment outright.
Under a resolution passed by House Democrats on the Rules Committee this past October, Trump and the White House potentially would have more rights to defend themselves in Judiciary Committee hearings. For example, attorneys for the president could participate in such hearings. But, in a bid for leverage, Chairman Jerrold Nadler, D-N.Y., would be allowed under the rules to deny \\"specific requests\\" by Trump representatives if the White House continued refusing to provide documents or witnesses sought by Democratic investigators.
A possible timetable for impeachment has been unclear. It’s generally thought the Judiciary Committee may hold a \\"markup\\" in which it writes articles of impeachment in mid-December. If that were to happen, it's possible the full House could vote on articles of impeachment sometime close to Christmas. That would be a similar timeframe to the impeachment of former President Bill Clinton: The House impeached Clinton just before Christmas in 1998. The Senate trial then began in January 1999.
However, the House theoretically could pass articles of impeachment, but delay a vote to send them to the Senate for consideration -- perhaps to delay handing Senate Majority Leader Mitch McConnell, R-Ky., control over the proceedings.
Graham, coming out of the White House discussion, added that \\"we didn't talk about\\" how to mount a case to \\"defend the president.\\"
Instead, Graham said, the discussion centered around \\"how would the trial start -- you know, they'll make a request for witnesses, but that would have to be granted by the Senate, I guess that's the way we did it before.\\"
Graham continued, \\"My preference was to try to follow the Clinton model as much as possible.\\"
Clinton was acquitted on both perjury and obstruction counts in February 1999, with each vote falling fall short of the two-thirds majority required for removal.
In the Senate, impeachment procedures would allow witnesses to be called by the president's defense lawyers, GOP senators and a team of House Democrats who essentially would serve as prosecutors. The big catch: Republicans would need enough votes from the 53 GOP senators to muster a majority and prevent Democrats from blocking them.
Assuming Republican senators would stay united -- not a guarantee -- Trump's defenders could try refocusing the inquiry by seeking testimony from people like Hunter Biden, the son of 2020 presidential hopeful Joe Biden.
During his July 25 call with Ukrainian President Volodymyr Zelensky that led to a whistleblower complaint touching off the impeachment inquiry, Trump suggested Zelensky investigate Joe Biden's dealings in Ukraine, including the former vice president's successful push to have Ukraine's top prosecutor fired by threatening to withhold $1 billion in U.S. aid while the prosecutor was investigating Burisma Holdings, where Hunter Biden served on the board.
Hunter Biden held that lucrative role despite limited expertise while his father oversaw Ukraine policy as vice president. If Senate Republicans could put forward evidence showing the president's concerns about the Bidens' potential corruption were legitimate, they could undercut Democrats' central argument for impeachment.
On Thursday, Graham strongly signaled that Hunter Biden would be a key GOP focus. He sent a letter to Secretary of State Mike Pompeo requesting documents \\"related to contacts between Vice President Biden, Hunter Biden, other Obama administration officials and Ukrainian President Petro Poroshenko.\\"
Deputy Assistant Secretary of State George Kent testified behind closed doors last month that he and other officials had qualms about Hunter Biden's lucrative role on the board of Burisma at the time.
\\"What Republicans want to do is broaden the story,\\" said David Hoppe, who was chief of staff to Senate Majority Leader Trent Lott, R-Miss., during Clinton's impeachment trial.
And, Sen. Kevin Cramer, R-N.D., said Thursday he'd like Senate testimony from the still-anonymous whistleblower, whose House appearance Democrats have blocked. Cramer said he might also like to hear from both Bidens and House Intelligence Committee Chairman Adam Schiff, D-Calif.
The White House has signaled it will mount an aggressive defense. \\"When this goes over to the Senate, you know, the people that actually started this thing, they are going to be put on the stand,\\" Eric Trump, the president's son, told reporters Thursday. He said that would include \\"heads of the Democratic Party.\\"
For his part, Trump has argued that U.S. Ambassador to the European Union Gordon Sondland's testimony before the Intelligence Committee was a total exoneration.\\"I just noticed one thing and that would mean it’s all over,\\" Trump said on the White House lawn before reading from handwritten notes taken during Sondland’s testimony. Sondland testified about a conversation with Trump during which he asked the president what he wanted from Ukraine.
\\"It was a very short, abrupt conversation,\\" the ambassador said. \\"He was not in a good mood, and he just said, 'I want nothing. I want nothing. I want no quid pro quo. Tell Zelensky to do the right thing.' Something to that effect.\\"",
  "href": "https://www.foxnews.com/politics/impeachment-articles-senate-trial-republicans",
  "publicationDate": "Wed Nov 20 2019 20:32:01 GMT-0600 (Central Standard Time)",
  "title": "Dems could draft 4 articles of impeachment, GOP plans for full Senate trial, sources say",
}
`;

exports[`fox snapshots gets rid of ^WATCH: type advertisements 1`] = `
Object {
  "authors": Array [
    Object {
      "href": "https://www.foxbusiness.com/person/g/jonathan-garber",
      "name": "Jonathan Garber",
    },
  ],
  "content": "The brokerage wars are heating up.
Charles Schwab is buying Ameritrade for $26 billion, sources tell FOX Business' Maria Bartiromo exclusively. TD Ameritrade shares soared on the news while Charles Schwab was higher.
The deal, expected to be announced this morning, will allow Charles Schwab to better compete with the likes of BlackRock. The combined firm will be run by Schwab CEO Walter Bettinger, and TD Ameritrade's chief financial officer, Steve Boyle, will lead the target company until the deal is completed.
TD Ameritrade's current CEO Tim Hockey, who announced in July that he would leave by the end of February, has acknowledged that his firm's decision to adopt zero commissions -- as Schwab and rivals E-Trade and Fidelity have done -- would prompt speculation about mergers.
\\"We will take a look at anything that makes financial and strategic sense,\\" he said last month, outlining the company's plans to make up for lost revenue of as much as $240 million a quarter from the new commission structure. \\"Scale is important. We have scale. We're very comfortable with our earnings power now, even in this new environment.\\"
Schwab said in October that eliminating its $4.95-per-trade commission would trim quarterly revenue by $90 million to $100 million, or about 3 percent to 4 percent of the total.
The firm's stock has risen 7.8 percent this year, while TD Ameritrade's has fallen 15.5 percent. The broader S&P 500 gained 24 percent in the same period.
TD Ameritrade, which began as First Omaha Securities, was founded by Joe Ricketts in1975. In 1988, it became the first firm to let clients make trades by touch-tone phone, then went public three years later. TD bought Scottrade in a $4 billion cash-and-stock deal in 2016, tying up two major players in the industry.",
  "href": "https://www.foxbusiness.com/markets/charles-schwab-buying-td-ameritrade-for-26b",
  "publicationDate": "Sun Nov 17 2019 20:32:01 GMT-0600 (Central Standard Time)",
  "title": "Charles Schwab buying TD Ameritrade for $26B: EXCLUSIVE",
}
`;
